### ISSUE TITLE
Resolve semantics of invoking `Object` members on JS objects and implement

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/sdk#35950]
An internal user hit the following trace when interoperating with non-Dart JS:

```
TypeError: obj[$toString] is not a function
    at Object.dart_library.library.dart.str
    ...
```

It looks like DDC is crashing here because there is no `[$toString]` on the prototype.  What should the behavior be (and it should be consistent)?

@jmesserly @natebosch @rakudrama

**Comment by matanlurey** [comment]
To clarify a bit, they caught an exception thrown by GWT, which is not a Dart exception.

It looks like DDC thought that the object had a _Dart_ `$toString` method (it did not).

**Comment by jmesserly** [comment]
I think this is a duplicate of https://github.com/dart-lang/sdk/issues/35951 (and https://github.com/dart-lang/sdk/issues/35952) ... going to merge this into those because I saw them first and added comments.

TL;DR -- we will likely want to fall back to JS `toString()`. That may be enough in this case, assuming GWT exposes a `toString` method on the prototype. Note that some JS objects do not support `toString`, and throw a TypeError if it is called.
